### PR TITLE
Add syscall/js.copyBytesToJS support

### PR DESCRIPTION
**Pull request by justinclift**

### CODE REVIEW

File: targets/wasm_exec.js [hunk 1]
```diff
@@ -319,6 +319,25 @@
...
+
+					// copyBytesToJS(dst ref, src []byte) (int, bool)
+					// Originally copied from upstream Go project, then modified:
+					//   https://github.com/golang/go/blob/3f995c3f3b43033013013e6c7ccc93a9b1411ca9/misc/wasm/wasm_exec.js#L404-L416
+					"syscall/js.copyBytesToJS": (ret_addr, dest_addr, source_addr, source_len, source_cap) => {
+						let num_bytes_copied_addr = ret_addr;
+						let returned_status_addr = ret_addr + 4; // Address of returned boolean status variable
+
+						const dst = loadValue(dest_addr);
+						const src = loadSlice(source_addr);
```
Comment (aykevl) [inline targets/wasm_exec.js]: I think you need to pass the `len` as well (`source_len`).

Looking at `loadSlice` it seems to ignore the `cap` parameter.

Comment (justinclift) [reply targets/wasm_exec.js]: Not sure.  Looking at the upstream Go version it's copied from, that one doesn't seem to use it?

https://github.com/golang/go/blob/3f995c3f3b43033013013e6c7ccc93a9b1411ca9/misc/wasm/wasm_exec.js#L404-L416

Comment (justinclift) [reply targets/wasm_exec.js]: Ahhh.  The TinyGo `loadslice()` defined earlier in the file.

Will update. :smile:

Comment (justinclift) [reply targets/wasm_exec.js]: Done.

Comment (aykevl) [reply targets/wasm_exec.js]: Thank you!

**Comment by aykevl** [comment]
Did you test this locally, to see that it at least works right now?

**Comment by justinclift** [comment]
Yep, it's working (even without the `source_len` tweak) for the older examples here I'm updating for newer TinyGo (with other TBD later):

* https://github.com/justinclift/tinygo-wasm-basic-triangle
* https://github.com/justinclift/tinygo-webgl-fundamentals-lesson-1

**Comment by aykevl** [comment]
Okay great, merged now.

**Pull request merged**